test/gate.go

```diff
@@ -0,0 +1,27 @@
+package test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/system"
+)
+
+// requireGate returns a setup func that will skip the current
+// test if the feature-flag with given name does not equal
+// given value. It will fatally fail the test if it cannot get
+// the feature-flag configmap.
+func requireGate(name, value string) func(context.Context, *testing.T, *clients, string) {
+	return func(ctx context.Context, t *testing.T, c *clients, namespace string) {
```

Comment:
namespace parameter seems not used. Is it here because required in the Setup() declaration?

Reply:
Yes you're exactly right - [the setup func expects functions with signature](https://github.com/tektoncd/pipeline/blob/879ba4f65732808a0614d76d5993af0bac736009/test/init_test.go#L52) matching:

```go
func(context.Context, *testing.T, *clients, string) // the string at the end here is namespace
```

Reply:
Neat !